library/std/src/process/tests.rs

```diff
@@ -417,6 +417,111 @@ fn env_empty() {
     assert!(p.is_ok());
 }
 
+#[test]
+#[cfg(not(windows))]
+#[cfg_attr(any(target_os = "emscripten", target_env = "sgx"), ignore)]
+fn main() {
+    const PIDFD: &'static str =
+        if cfg!(target_os = "linux") { "    create_pidfd: false,\n" } else { "" };
+
+    let mut command = Command::new("some-boring-name");
+
+    assert_eq!(format!("{command:?}"), format!(r#""some-boring-name""#));
+
+    assert_eq!(
+        format!("{command:#?}"),
+        format!(
+            r#"Command {{
+    program: "some-boring-name",
+    args: [
+        "some-boring-name",
+    ],
+    env: CommandEnv {{
+        clear: false,
+        saw_path: false,
+        vars: {{}},
+    }},
+    cwd: None,
+    uid: None,
+    gid: None,
+    saw_nul: false,
+    groups: None,
+    pgroup: None,
+{PIDFD}}}"#
+        )
+    );
+
+    command.args(&["1", "2", "3"]);
+
+    assert_eq!(format!("{command:?}"), format!(r#""some-boring-name" "1" "2" "3""#));
+
+    assert_eq!(
+        format!("{command:#?}"),
+        format!(
+            r#"Command {{
+    program: "some-boring-name",
+    args: [
+        "some-boring-name",
+        "1",
+        "2",
+        "3",
+    ],
+    env: CommandEnv {{
+        clear: false,
+        saw_path: false,
+        vars: {{}},
+    }},
+    cwd: None,
+    uid: None,
+    gid: None,
+    saw_nul: false,
+    groups: None,
+    pgroup: None,
+{PIDFD}}}"#
+        )
+    );
+
+    crate::os::unix::process::CommandExt::arg0(&mut command, "exciting-name");
+
+    assert_eq!(
+        format!("{command:?}"),
+        format!(r#"["some-boring-name"] "exciting-name" "1" "2" "3""#)
+    );
+
+    assert_eq!(
+        format!("{command:#?}"),
+        format!(
+            r#"Command {{
+    program: "some-boring-name",
+    args: [
+        "exciting-name",
+        "1",
+        "2",
+        "3",
+    ],
+    env: CommandEnv {{
+        clear: false,
+        saw_path: false,
+        vars: {{}},
+    }},
+    cwd: None,
+    uid: None,
+    gid: None,
+    saw_nul: false,
+    groups: None,
+    pgroup: None,
+{PIDFD}}}"#
+        )
+    );
+
+    let mut command_with_env_and_cwd = Command::new("boring-name");
+    command_with_env_and_cwd.current_dir("/some/path").env("FOO", "bar");
+    assert_eq!(
+        format!("{command_with_env_and_cwd:?}"),
+        r#"cd "/some/path" && "FOO"="bar" "boring-name""#
```

Comment:
By construction env variables are quoted (because using the debug impl of `OsString`. I don't remember if that's fine for env variables or if the quotes will be included in the value

Reply:
Hrm no, this doesn't work in bash at least. The value can be quoted but not the key.
You can try `to_str()` and then use the Display impl.

Handling non-utf8 chars would need additional escaping machinery... I think we can skip that for now, debugging is a best-effort thing anyway.

Reply:
I went with `to_str_lossy` since it's a debug impl